ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -453,11 +485,9 @@ private void close(SelectionKey key) {
       this.disconnected.add(transmission.getConnectionId());
       this.keyMap.remove(transmission.getConnectionId());
       activeConnections.set(this.keyMap.size());
-      try {
-        transmission.close();
-      } catch (IOException e) {
-        logger.error("IOException thrown during closing of transmission with connectionId {} :",
-            transmission.getConnectionId(), e);
+      transmission.close();
+      if (pendingSslTrans.remove(transmission.getConnectionId()) != null) {
```

Comment:
would prefer doing this before `transmission.close()`. Ideally `close` is the last thing you want to do with an object (what if it cleared its state?).